### PR TITLE
Support capturing http headers

### LINKED ISSUE #1693: Log HTTP Headers, Request/Response Body as Custom Request Properties
**Is your feature request related to a problem? Please describe.**
We use extensive custom HTTP Headers for various API transactions and would like to trace or have visibility into App Insights and visualize as part of the Request Properties panel

**Describe the solution you would like**
Allow configuration to enable logging for list of HTTP headers specified or if possible log all headers, Request/Response Body as custom Request Properties

**Describe alternatives you have considered**
Currently, we are trying to print headers, body as an INFO in STDOut to get captured in detailed telemetry and then locate transactions via transaction search

**Additional context**
Current Use: Java Agent - 3.0.3 GA
Request Properties Panel - Custom Properties

image

### PR DESCRIPTION
**Pull request by trask**
Resolves #1693

### CODE REVIEW

File: test/smoke/appServers/global-resources/httpheaders_applicationinsights.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{
+  "connectionString": "InstrumentationKey=00000000-0000-0000-0000-0FEEDDADBEEF;IngestionEndpoint=http://host.docker.internal:6060/",
+  "role": {
+    "name": "testrolename",
+    "instance": "testroleinstance"
+  },
+  "preview": {
+    "captureHttpServerHeaders": {
```
Comment (heyams) [inline test/smoke/appServers/global-resources/httpheaders_applicationinsights.json:8]: not sure how server headers will be helpful for customers?  can you share a use case for this?

Comment (heyams) [reply test/smoke/appServers/global-resources/httpheaders_applicationinsights.json:8]: is it customer's server or app insights backend server? it's not clear to me.

**Review by heyams — APPROVED** [review]

**Comment by zoomingrocket** [comment]
@trask  - How can we use this with 3.2.5-BETA? What should be configured in applicationinsights.json config?
We tried below but does not seem to work

`  "preview": {
    "instrumentation": {
      "vertx": {
        "enabled": true
      },
      "captureHttpServerHeaders": {
        "enabled": true
      }
`

**Comment by trask** [comment]
hey @zoomingrocket, check out https://docs.microsoft.com/en-us/azure/azure-monitor/app/java-standalone-config#http-headers

**Pull request merged**